compiler/src/dotty/tools/dotc/Driver.scala

```diff
@@ -64,19 +64,33 @@ class Driver {
 
   protected def sourcesRequired: Boolean = true
 
-  def setup(args: Array[String], rootCtx: Context): (List[AbstractFile], Context) = {
+  protected def command: CompilerCommand = ScalacCommand
+
+  /** Setup context with initialized settings from CLI arguments, then check if there are any settings that
+   *  would change the default behaviour of the compiler.
+   *
+   *  @return A tuple of Option of List and Context.
+   *  If there is no setting preventing us from continuing compilation, function returns Some(List)
+   *  and Context updated respectively with compilation files. In particular, it can be Some(Nil).
+   *  If compilation should be interruped, function returns None and an old Context.
```

Comment:
```suggestion
   *  If compilation should be interrupted, this method returns None and an old Context.
```
But if you're returning the original Context, then there's no point in returning it at all and you could change the result type of this method to `Option[(List[AbstractFile], Context)]`

Reply:
Good point, will change